tests/test_databases.py

```diff
@@ -0,0 +1,53 @@
+import io
+import os
+import json
+from .base import BaseTinyFatTest
+from tempfile import NamedTemporaryFile
+from tinyfatdb import TinyFatDB, TinyFatTable
+from tinydb.storages import MemoryStorage
+
+
+########################################################################
+class TestCreateDB(BaseTinyFatTest):
+    """
+    Tests to ensure creating, loading, and writing to database work for
+    file-based dbs.
+    """
+
+    ####################################################################
+    def setUp(self):
+        data = {
+            "ABC": {
+                "1": {"a": 1}
+            }
+        }
+        self.json_file = NamedTemporaryFile(suffix=".json", delete=False)
+        self.json_file.write(json.dumps(data).encode("utf-8"))
+        self.json_file.flush()
+        self.json_file.close()
+
+    ####################################################################
+    def tearDown(self):
+        os.unlink(self.json_file.name)
+
+    ####################################################################
+    def test_create_db__manual(self):
+        db = TinyFatDB(default_table="ABC", table_class=TinyFatTable, storage=MemoryStorage)
+        table = db.table("ABC")
+        self.assertTrue(isinstance(table, TinyFatTable))
+        self.assertTrue(isinstance(db._storage, MemoryStorage))
+
+    ####################################################################
+    def test_create_db__json(self):
+        db = TinyFatDB.create(self.json_file.name, name="ABC", table=TinyFatTable)
```

Comment:
One thing to note is that I got rid of the `ABCModel`/`FooModel` that you were using. Perhaps I misunderstood, but it seems that the only purpose for those models was to use the functions `sum` and `longest`. Farther down I will comment on the tests that were using them and why I think that it would be good to steer away from them.

With those two functions not being utilized, there was not a need to have them. I converted all occurrences to the `TinyFatTable` class.